### PR TITLE
Feat: support query service endpoints by velaQL

### PR DESCRIPTION
**Pull request by barnettZQG**
Signed-off-by: barnettZQG <barnett.zqg@gmail.com>

### Description of your changes

Generate access endpoint information for the application. By reading Ingress and Service resources. Support common component type and helm component.

I have:

- [x] Read and followed KubeVela's [contribution process](https://github.com/oam-dev/kubevela/blob/master/contribute/create-pull-request.md).
- [x] [Related Docs](https://github.com/oam-dev/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### Special notes for your reviewer

/cc @wangyikewxgm

### CODE REVIEW

**Review by wonderflow — COMMENTED** [review]
we should use velaQL to query information

File: pkg/velaql/providers/query/handler.go [hunk 1]
```diff
@@ -154,6 +164,96 @@ func (h *provider) SearchEvents(ctx wfContext.Context, v *value.Value, act types
...
+		return fmt.Errorf("query app failure %w", err)
+	}
+	var serviceEndpoints []common.ServiceEndpoint
+	for _, resource := range app.Status.AppliedResources {
+		if !isResourceInTargetCluster(opt.Filter, resource) {
+			continue
+		}
+		switch resource.Kind {
+		case "Ingress":
+			if resource.GroupVersionKind().Group == "networking.k8s.io" && (resource.GroupVersionKind().Version == "v1beta1" || resource.GroupVersionKind().Version == "v1") {
```
Comment (yangsoon) [inline pkg/velaql/providers/query/handler.go:206]:
```suggestion
			if resource.GroupVersionKind().Group == networkv1.GroupName && (resource.GroupVersionKind().Version == "v1beta1" || resource.GroupVersionKind().Version == "v1") {
```

File: apis/core.oam.dev/common/types.go [hunk 2]
```diff
@@ -325,6 +327,53 @@ type PolicyStatus struct {
...
+	Host string `json:"host"`
+
+	// the port for the endpoint
+	// Default is 80.
+	Port int32 `json:"port"`
+
+	// the path for the endpoint
+	Path string `json:"path,omitempty"`
+}
+
```
Comment (yangsoon) [inline apis/core.oam.dev/common/types.go:376]: These types are not related to OAM and can be moved to the velaql folder

**Review by wonderflow — APPROVED** [review]
good job

**Review by FogDong — APPROVED** [review]
